builtin/builtin.go

```diff
@@ -189,6 +189,14 @@ func builtin_print(self py.Object, args py.Tuple, kwargs py.StringDict) (py.Obje
 	end := endObj.(py.String)
 	// FIXME ignoring file and flush
 	for i, v := range args {
+                switch sv := v.(type) {
+                    case py.I__str__:
+                        v, _ = sv.M__str__()
```

Comment:
This isn't quite right - this needs to call `py.Str` which will fall back to py.Repr

If you only look for the `M__str__` then you'll miss the user defined methods.

Reply:
This works with this test:

    class X(object):
      def __init__(self, x):
        self.x = x
      def __str__(self):
        return "my name is %s" % self.x

    class Y(object):
      def __init__(self, x):
        self.x = x

      def __repr__(self):
        return "my nome is %s" % self.x

    print(X(42))
    print(Y('hello'))

Note that I cannot add any test to builtin/tests/builtin.py because there is no way to check the result of print (until support for redirects is implemented)

Reply:
re-implemented with py.Str